lib/Base/Image.cpp

```diff
@@ -391,9 +391,32 @@ Tensor glow::readPngImageAndPreprocess(llvm::StringRef filename,
     imageData.transpose(&transposed, {2u, 0u, 1u});
     imageData = std::move(transposed);
   }
+}
+
+Tensor glow::readPngImageAndPreprocess(llvm::StringRef filename,
+                                       ImageNormalizationMode imageNormMode,
+                                       ImageChannelOrder imageChannelOrder,
+                                       ImageLayout imageLayout,
+                                       llvm::ArrayRef<float> mean,
+                                       llvm::ArrayRef<float> stddev) {
+  Tensor imageData;
+  readPngImageAndPreprocessHelper(imageData, filename, imageNormMode,
+                                  imageChannelOrder, imageLayout, mean, stddev);
   return imageData;
 }
 
+void glow::readPngImageAndPreprocess(Tensor &imageData,
```

Comment:
Given that this version of `readPngImageAndPreprocess()` is just forwarding to `readPngImageAndPreprocessHelper()` without any extra logic, I think it's preferable to avoid adding `readPngImageAndPreprocessHelper()`. You can just call this version of `readPngImageAndPreprocess()` (which takes a Tensor reference) from the other version which returns the Tensor.

Reply:
sounds good